arangod/Auth/TokenCache.cpp

```diff
@@ -327,6 +327,23 @@ auth::TokenCache::Entry auth::TokenCache::validateJwtBody(std::string const& bod
     return auth::TokenCache::Entry::Unauthenticated();
   }
 
+  if (bodySlice.hasKey("allowed_paths")) {
+    VPackSlice const paths = bodySlice.get("allowed_paths");
+    if (!paths.isArray()) {
+      LOG_TOPIC(TRACE, arangodb::Logger::AUTHENTICATION)
+        << "allowed_paths must be an array";
+      return auth::TokenCache::Entry::Unauthenticated();
+    }
+    for (auto const& path : VPackArrayIterator(paths)) {
```

Comment:
Shall we handle the case when "allowed_paths" is an empty array?
In that particular case, access to *all* paths will be allowed, but I find that unintuitive. Instead, we should insist that if "allowed_paths" is specified, it at least contains one entry. Otherwise we cannot distinguish between "allowed_paths" being present but empty, and "allowed_paths" not being present.

Reply:
Fixed by disallowing empty `allowed_paths`.